src/frontend/src/server.rs

```diff
@@ -178,48 +177,31 @@ impl Services {
             }
             http_server.set_script_handler(instance.clone());
 
-            Some((Box::new(http_server) as _, http_addr))
-        } else {
-            None
-        };
+            result.insert(HTTP_SERVER.to_string(), (Box::new(http_server), http_addr));
+        }
 
-        let prom_server_and_addr = if let Some(prom_options) = &opts.prom_options {
+        if let Some(prom_options) = &opts.prom_options {
             let prom_addr = parse_addr(&prom_options.addr)?;
 
-            let mut prom_server = PromServer::create_server(instance.clone());
+            let mut prom_server = PromServer::create_server(instance);
             if let Some(user_provider) = user_provider {
                 prom_server.set_user_provider(user_provider);
             }
 
-            Some((prom_server as _, prom_addr))
-        } else {
-            None
+            result.insert(PROM_SERVER.to_string(), (prom_server, prom_addr));
         };
-
-        try_join!(
-            start_server(http_server_and_addr),
-            start_server(grpc_server_and_addr),
-            start_server(mysql_server_and_addr),
-            start_server(postgres_server_and_addr),
-            start_server(opentsdb_server_and_addr),
-            start_server(prom_server_and_addr),
-        )
-        .context(error::StartServerSnafu)?;
-        Ok(())
+        Ok(result)
     }
 }
 
 fn parse_addr(addr: &str) -> Result<SocketAddr> {
     addr.parse().context(error::ParseAddrSnafu { addr })
 }
 
-async fn start_server(
-    server_and_addr: Option<(Box<dyn Server>, SocketAddr)>,
+pub async fn start_server(
+    server_and_addr: &(Box<dyn Server>, SocketAddr),
 ) -> servers::error::Result<Option<SocketAddr>> {
-    if let Some((server, addr)) = server_and_addr {
-        info!("Starting server at {}", addr);
-        server.start(addr).await.map(Some)
-    } else {
-        Ok(None)
-    }
+    let (server, addr) = server_and_addr;
+    info!("Starting server at {}", addr);
```

Comment:
If we move the server name to the `Server` trait, then we can print the server name to log too.